test/unit/math/prim/mat/fun/add_diag_test.cpp

```diff
@@ -59,29 +59,29 @@ TEST(MathPrimMat, double_mat_double_rvec_add_diag) {
 }
 
 TEST(MathPrimMat, var_mat_double_vec_add_diag) {
-  Eigen::Matrix<stan::math::var, Eigen::Dynamic, Eigen::Dynamic> mat(2, 3);
+  Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> mat(2, 3);
   mat << 1, 1, 1, 1, 1, 1;
 
   Eigen::Matrix<double, 1, -1> to_add(2);
   to_add << 0, 1;
 
-  Eigen::Matrix<stan::math::var, Eigen::Dynamic, Eigen::Dynamic> out_mat;
+  Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> out_mat;
   EXPECT_NO_THROW(out_mat = stan::math::add_diag(mat, to_add));
   for (int i = 0; i < 2; ++i)
-    EXPECT_FLOAT_EQ(1 + to_add[i], value_of(out_mat(i, i)))
+    EXPECT_FLOAT_EQ(1 + to_add[i], out_mat(i, i))
         << "index: ( " << i << ", " << i << ")";
 }
 
 TEST(MathPrimMat, var_mat_double_rvec_add_diag) {
-  Eigen::Matrix<stan::math::var, Eigen::Dynamic, Eigen::Dynamic> mat(2, 3);
+  Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> mat(2, 3);
   mat << 1, 1, 1, 1, 1, 1;
 
   Eigen::Matrix<double, -1, 1> to_add(2);
   to_add << 0, 1;
 
-  Eigen::Matrix<stan::math::var, Eigen::Dynamic, Eigen::Dynamic> out_mat;
+  Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> out_mat;
   EXPECT_NO_THROW(out_mat = stan::math::add_diag(mat, to_add));
   for (int i = 0; i < 2; ++i)
-    EXPECT_FLOAT_EQ(1 + to_add[i], value_of(out_mat(i, i)))
+    EXPECT_FLOAT_EQ(1 + to_add[i], out_mat(i, i))
```

Comment:
Checking that the off-diagonals don't change as well would make this a better test.

In super pseudo-code:
```
for(int i...) {
  for(int j...) {
    if(i == j) {
      check_diagnonal(i);
    } else {
      check_off_diagonal(i, j);
    }
  }
}
```